Sources/NIOCore/ByteBuffer-views.swift

```diff
@@ -158,6 +158,27 @@ extension ByteBufferView: RangeReplaceableCollection {
         self = ByteBufferView(ByteBuffer())
     }
 
+    /// Reserves enough space in the underlying `ByteBuffer` to store the specified
+    /// number of bytes.
+    ///
+    /// See the documentation for ``ByteBuffer.reserveCapacity(_:)`` for more details.
+    @inlinable
+    public mutating func reserveCapacity(_ minimumCapacity: Int) {
+        self._buffer.reserveCapacity(minimumCapacity)
```

Comment:
We don't necessarily care here, but `ByteBuffer`'s `reserveCapacity` method doesn't have exactly the semantics we want. Specifically it completely disregards the reader/writer indices, so if this `ByteBufferView` is a slice of a larger buffer, we need to be a little careful. In this instance, we probably want to do something like this:

```swift
let currentBufferCapacity = self._buffer.capacity
let currentViewSize = self.count
let requestedExtraCapacity = max(minimumCapacity - currentViewSize, 0)
self._buffer.reserveCapacity(currentBufferCapacity + requestedExtraCapacity)
```